suitcase/structure.py

```diff
@@ -85,6 +86,11 @@ def unpack_stream(self, stream):
         logic present for dealing with checksum fields.
 
         """
+        def is_substructure(field):
+            return isinstance(field, SubstructureField) or \
+                    isinstance(field, ConditionalField) and \
+                    isinstance(field.field, SubstructureField)

```

Comment:
The other thing is that there are other fields that could wrap `SubstructureField` such as `LengthField` that should have similar logic.  I'm fixing this with the change to use polymorphism (i.e. `field.is_substructure()`).